#! /bin/sh
patch -p1 -f $* < $0
exit $?

We really want to build newlib.  Ignore the cross compiling check.



diff -upr newlib/configure.ac newlib-1.16.0/configure.ac
--- newlib/configure.ac	2008-01-17 16:50:24.000000000 +0100
+++ newlib-1.16.0/configure.ac	2007-12-18 08:37:34.000000000 +0100
@@ -263,7 +263,7 @@
 
 # Similarly, some are only suitable for cross toolchains.
 # Remove these if host=target.
-cross_only="target-libgloss target-newlib target-opcodes"
+cross_only="target-libgloss target-opcodes"
 
 case $is_cross_compiler in
   no) skipdirs="${skipdirs} ${cross_only}" ;;
@@ -1666,10 +1666,6 @@
 # Handle --with-headers=XXX.  If the value is not "yes", the contents of
 # the named directory are copied to $(tooldir)/sys-include.
 if test x"${with_headers}" != x && test x"${with_headers}" != xno ; then
-  if test x${is_cross_compiler} = xno ; then
-    echo 1>&2 '***' --with-headers is only supported when cross compiling
-    exit 1
-  fi
   if test x"${with_headers}" != xyes ; then
     x=${gcc_cv_tool_prefix}
     copy_dirs="${copy_dirs} ${with_headers} $x/${target_noncanonical}/sys-include"

